### PR TITLE
refactor(slug): Default input to sane

### PR DESCRIPTION
**Pull request by AnthonyWadham**
# feat: Add SlugToSaneDefault helper function for project slug sanitization

Closes: WORLD-XXX

## Overview

This PR adds a new helper function `SlugToSaneDefault` that sanitizes input strings into a standardized slug format, and applies it to project slugs during creation.

## Brief Changelog

- Added `SlugToSaneDefault` helper function in `common.go` that:
  - Converts input to lowercase
  - Trims whitespace
  - Replaces spaces with hyphens
  - Removes non-alphanumeric characters (except hyphens)
  - Replaces multiple consecutive hyphens with a single hyphen
  - Trims hyphens from start and end
- Applied this function to project slugs in `inputProjectSlug` to ensure consistent formatting

## Testing and Verifying

This change is a trivial rework/code cleanup without any test coverage. The function implements standard slug sanitization practices that can be manually verified by creating projects with various input formats.

## Greptile Summary

Added a new helper function `SlugToSaneDefault` to standardize project slug formatting, with automatic sanitization of input strings to lowercase alphanumeric characters with hyphens.

- Added `SlugToSaneDefault` in `cmd/world/forge/common.go` for consistent slug formatting
- Applied sanitization in `cmd/world/forge/project.go` before slug validation
- Function converts input to lowercase, replaces spaces with hyphens, removes special chars
- Potential concern: Order of operations between sanitization and validation could cause issues
- Recommend adding test coverage to verify behavior with edge cases and non-ASCII characters

## Summary by CodeRabbit

- **New Features**
  - Enhanced slug input handling with automatic sanitization and normalization, improving consistency and reducing errors during organization and project creation.

### CODE REVIEW

**Comment by Doglightning** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/129?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#129** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/129?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/129?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#128** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/128?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#127** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/127?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite/merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/Argus-Labs?ref=merge-queue-instructions-in-stack-comment&prId=129).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: cmd/world/forge/common.go [hunk 1]
```diff
@@ -295,6 +295,16 @@ func checkLogin() bool {
 	return true
 }
 
+// SlugToSaneDefault is helper function to ensure input is sane default.
+func SlugToSaneDefault(input string) string {
+	slug := strings.ToLower(strings.TrimSpace(input))
+	slug = strings.ReplaceAll(slug, " ", "_")
+	slug = regexp.MustCompile(`[^a-z0-9_]`).ReplaceAllString(slug, "")
+	slug = regexp.MustCompile(`_+`).ReplaceAllString(slug, "_")
+	slug = strings.Trim(slug, "_")
```
[the remark this answers is not in the document]
Comment (Doglightning) [reply cmd/world/forge/common.go:304]: I was just following the slugCheck function below "Slug can only contain lowercase letters, numbers, and underscores" which didnt accept hyphens.  

Can change both to hyphens instead of underscore if required

[the remark this answers is not in the document]
Comment (ezavada) [reply cmd/world/forge/common.go:304]: Nope, underscores are what is allowed, no hypens

**Review by ezavada — APPROVED** [review]
